nilearn/reporting/html_report.py

```diff
@@ -0,0 +1,188 @@
+    body = tpl.substitute(title=title, content=content,
+                          overlay=overlay,
+                          docstring=docstring,
+                          parameters=parameters,
+                          description=description)
+
+    head_template_name = 'report_head_template.html'
+    head_template_path = resource_path.joinpath(head_template_name)
+    with open(str(head_template_path), 'r') as head_file:
+        head_tpl = Template(head_file.read())
+
+    return HTMLReport(body=body, head_tpl=head_tpl)
+
+
+class ReportMixin:
+    """
+    A class to provide general reporting functionality
+    """
+
+    def _define_overlay(self):
+        """
+        Determine whether an overlay was provided and
+        update the report text as appropriate.
+
+        Parameters
+        ----------
+
+        Returns
+        -------
+        """
+        displays = self._reporting()
+
+        if len(displays) == 1:  # set overlay to None
+            overlay, image = None, displays[0]
+
+        elif len(displays) == 2:
+            overlay, image = displays[0], displays[1]
+
+        return overlay, image
+
+    def generate_report(self):
+        """
+        Generate a report for Nilearn objects.
+
+        Report is useful to visualize steps in a processing pipeline.
+        Example use case: visualize the overlap of a mask and reference image
+        in NiftiMasker.
+
+        Returns
+        -------
+        report : HTMLDocument
+        """
+        if not hasattr(self, 'input_'):
+            warnings.warn('Report generation not enabled !'
+                          'No visual outputs will be created.')
+            report = _update_template(title='Empty Report',
+                                      docstring=('This report was not '
+                                                 'generated. Please check '
+                                                 'that reporting is enabled.'),
+                                      content=_embed_img(None),
+                                      overlay=_embed_img(None),
+                                      parameters=dict())
+
+        else:
+            overlay, image = self._define_overlay()
+            description = self._report_description
+            parameters = _str_params(self.get_params())
+            docstring = self.__doc__
+            snippet = docstring.partition('Parameters\n    ----------\n')[0]
+            report = _update_template(title=self.__class__.__name__,
+                                      docstring=snippet,
+                                      content=_embed_img(image),
+                                      overlay=_embed_img(overlay),
+                                      parameters=parameters,
+                                      description=description)
+        return report
+
+
+class HTMLReport(plot_utils.HTMLDocument):
+    """
+    A report written as HTML.
+    Methods such as save_as_html(), open_in_browser()
```

Comment:
to respect the interface of the parent class could you add an `html` property that calls `get_standalone`? we cannot remove the `html` attribute from the parent class for backward compatibility

Reply:
:heavy_check_mark: This is done ! Let me know if it's not what you had in mind ?

Reply:
I was thinking something like

```
@property
def html(self):
    return self.get_standalone()
```

Reply:
yes, we settled on the solution proposed here
https://github.com/nilearn/nilearn/pull/2019#issuecomment-527956538

so it is addressed by initializing the HTMLReport with the correct html